### PR TITLE
chore: Internal build flavor

### PR DESCRIPTION
**Pull request by MohamadJaara**
* build internal green app

* rename internal to beta and add an internal build flavour

* rename internal to beta in Jenkinsfile and add the correct suffix

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [ ] contains a reference JIRA issue number like `SQPIT-764`
    - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

we are missing a build flavor to replace the internal (green icon)

### Solutions

1. rename wire internal build flavor to Beta.
2. add internal that in scala compatible.
3. add new branch `internal` for the new flavor.
4. clean up `variants_gradle.kts`.
5. change JenkinsFile to upload to play store only when pushing to main

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_ 

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-1302/2/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(5431062)(5431062)-dev-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/compat/com.wire.android-vv0.0.1(5431062)(5431062)-dev-compat.apk)

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-1302/4/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(5431279)(5431279)-dev-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/compat/com.wire.android-vv0.0.1(5431279)(5431279)-dev-compat.apk)

File: Jenkinsfile [hunk 1]
```diff
@@ -365,7 +371,7 @@ pipeline {
           }
           stage('Playstore') {
             when {
-              expression { env.trackName != 'None' && env.flavor != 'Dev' && env.CHANGE_ID == null }
+              expression { env.trackName != 'None' && env.flavor == 'Beta' && env.CHANGE_ID == null }
```
Comment (MohamadJaara) [inline Jenkinsfile:376]: When pushing, jerkins tried to publish to app store even if the flavor is not Beta

File: Jenkinsfile [hunk 2]
```diff
@@ -21,21 +24,24 @@ def defineBuildType() {
...
+    // use the scala client signing keys for testing upgrades.
+    return "Compat"
 }
 
 def defineTrackName() {
     def overwrite = env.CUSTOM_TRACK
     if(overwrite != null) {
         return overwrite
+    } else if (branchName == "main") {
+        return 'internal'
```
Comment (MohamadJaara) [inline Jenkinsfile:44]: set playstore track name only when pushing to main

File: Jenkinsfile [hunk 3]
```diff
@@ -6,12 +6,15 @@ def defineFlavor() {
     }
 
     def branchName = env.BRANCH_NAME
+
     if (branchName == "main") {
-        return 'Internal'
+        return 'Beta'
```
Comment (MohamadJaara) [inline Jenkinsfile:11]: i did rename what we call internal (wire beta) to Beta and added internal flavor to replace wire internal with green icon

File: buildSrc/src/main/kotlin/scripts/variants.gradle.kts [hunk 4]
```diff
@@ -20,30 +21,41 @@ object BuildTypes {
...
-    const val INTERNAL = "internal"
-    const val PUBLIC = "public"
-    const val FDROID = "fdroid"
-    const val STAGING = "staging"
-}
+sealed class ProductFlavors(
+    val applicationId: String,
+    val buildName: String,
+    val applicaitonIdSuffix: String? = null,
+    val dimentions: String = FlavorDimensions.DEFAULT
```
Comment (yamilmedina) [inline buildSrc/src/main/kotlin/scripts/variants.gradle.kts]:
```suggestion
    val dimensions: String = FlavorDimensions.DEFAULT
```

File: buildSrc/src/main/kotlin/scripts/variants.gradle.kts [hunk 5]
```diff
@@ -20,30 +21,41 @@ object BuildTypes {
...
-    const val DEV = "dev"
-    const val INTERNAL = "internal"
-    const val PUBLIC = "public"
-    const val FDROID = "fdroid"
-    const val STAGING = "staging"
-}
+sealed class ProductFlavors(
+    val applicationId: String,
+    val buildName: String,
+    val applicaitonIdSuffix: String? = null,
```
Comment (yamilmedina) [inline buildSrc/src/main/kotlin/scripts/variants.gradle.kts]:
```suggestion
    val applicationIdSuffix: String? = null,
```

**Review by yamilmedina — APPROVED** [review]
Small typos easy fixes, great work 👏  much sense now with our release tracks

File: default.json [hunk 6]
```diff
@@ -55,6 +55,25 @@
         "fcmProjectID": "wiregcmpushstaging",
         "isStaging": true
     },
+    "beta": {
+        "applicationId": "com.wire.internal",
+        "userId": "",
+        "_comment": "appName must match the value of 'app_name' in strings.xml",
+        "appName": "Wire Internal",
+        "launcherIcon": "ic_launcher_wire_internal",
```
Comment (yamilmedina) [inline default.json:63]: I think we'll need to add this resource too =D

File: buildSrc/src/main/kotlin/scripts/variants.gradle.kts [hunk 7]
```diff
@@ -99,21 +111,10 @@ android {
...
-        }
-        create(ProductFlavors.INTERNAL) {
-            dimension = FlavorDimensions.DEFAULT
-            applicationIdSuffix = ".${ProductFlavors.INTERNAL}"
-            versionNameSuffix = "-${ProductFlavors.INTERNAL}"
-        }
+        createAppFlavour(ProductFlavors.Dev)
+        createAppFlavour(ProductFlavors.Staging)
+        createAppFlavour(ProductFlavors.Beta)
+        createAppFlavour(ProductFlavors.Internal)
```
Comment (vitorhugods) [inline buildSrc/src/main/kotlin/scripts/variants.gradle.kts:117]: 🆒

**Review by vitorhugods — COMMENTED** [review]
Code-wise, it looks pretty good.

Just one question... what exactly is the intent here?

Are we going to start publishing to `internal` too? If so, when? What's the release cycle for `internal` builds?

Process-wise, I'm starting to feel confused.

I would love to have this `internal` to be published in every `develop` update, like what we have set for the `Dev` flavour that isn't published at all.

File: default.json (hunk 6 above)
Comment (MohamadJaara) [reply default.json:63]: most of the values in this file are not used and simply copied from the old app, i will raise another PR with clean up

**Comment by MohamadJaara** [comment]
@vitorhugods 
1. I wanted to make it easier to add new flavors.
2. atm I wanted to push to internal manually to organize a release pilot, for the migration from scala.
3. not sure if we want to push to internal for every dev since afaik it is used internally.

**Comment by AndroidBob** [comment]
Build [11](https://10.10.124.134/job/wire-android-reloaded/job/PR-1302/11/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(5431814)(5431814)-dev-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/compat/com.wire.android-vv0.0.1(5431814)(5431814)-dev-compat.apk)

**Pull request merged**